### PR TITLE
Add unique identifier for Price

### PR DESCRIPTION
**Pull request by tompntn**
This PR introduces the concept of "unique identifier for a price". This is a globally unique 256-bit number representing an individual price.

### CODE REVIEW

**Review by jayantk — APPROVED** [review]

**Comment by ali-bahjati** [comment]
Notes on the Solana change:

TD;LR: Solana API is not very clean and implemented solution follows patterns that are used by solana developers in famous projects.

AccountInfo in Solana is for on-chain programs and Account is for off-chain programs. Although they share many fields, they are not easily convertible to each other and the conversion is not obvious. So a helper function is provided for that. 

Alternatives considered:
- Creating Price from Account directly: it requires the library to include `solana-sdk` package as dependency which prevents package to be compiled in bpf target. To work around a feature flag is needed and it makes things very complex. (e.g: Examples need feature to be compiled and whenever you run `cargo test` you should pass proper features)
- Having  `load_price(price_key: &Pubkey, data: &[u8])`: Although this is a very nice API it will move ugly stuff from Solana inside the lib. So even in the on-chain program people should think about the key. Also it is not a common pattern within solana developers.

File: pyth-sdk-solana/README.md [hunk 1]
```diff
@@ -47,18 +47,31 @@ The examples below assume that the user has already obtained this account data.
...
 You can find price accounts in the pyth.network website (e.g.: [Crypto.BTC/USD accounts](https://pyth.network/markets/#Crypto.BTC/USD)).  
 
-To read the price from a price account, this library provides a `load_price` method that translates the binary account data into a `Price` struct:
+The `Price` struct contains several useful functions for working with the price.
+Some of these functions are described below.
+For more detailed information, please see the crate documentation.
+
+#### On-chain
+
+To read the price from a price account on-chain, this library provides a `load_price_from_account_info` method that constructs `Price` struct from Account Info:
```
Comment (jayantk) [inline pyth-sdk-solana/README.md:56]: `AccountInfo`

**Review by jayantk — APPROVED** [review]

File: pyth-sdk-solana/examples/eth_price.rs [hunk 2]
```diff
@@ -19,8 +19,8 @@ fn main() {
 
     loop {
         // get price data from key
-        let eth_price_data = clnt.get_account_data(&eth_price_key).unwrap();
-        let eth_price = load_price(&eth_price_data).unwrap();
+        let mut eth_price_account = clnt.get_account(&eth_price_key).unwrap();
+        let eth_price = load_price_from_account(&eth_price_key, &mut eth_price_account).unwrap();
```
Comment (tompntn) [inline pyth-sdk-solana/examples/eth_price.rs:23]: Should the Solana SDK expose a convenience function wrapping this, which given a client and a price key, returns a price? Probably a nice to have though 😄

Comment (ali-bahjati) [reply pyth-sdk-solana/examples/eth_price.rs:23]: If we get involved with client in the sdk then it requires a lot of dirty features to handle compiling in bpf. So essentially let's say for now it's sdk not a client. Maybe in the future we can consider a separate client package.

File: pyth-sdk-solana/README.md (hunk 1 above)
Comment (ali-bahjati) [reply pyth-sdk-solana/README.md]: Fixed.

**Pull request merged**